lib/workers/testbot.ts

```diff
@@ -0,0 +1,278 @@
+import * as sdk from 'etcher-sdk';
+import * as retry from 'bluebird-retry';
+import * as visuals from 'resin-cli-visuals';
+import { getDrive } from '../helpers';
+import { fs } from 'mz';
+import { Mutex } from 'async-mutex';
+import { EventEmitter } from 'events';
+import { Signer } from 'crypto';
+
+/**
+ * TestBot Hardware config
+ */
+const DEV_ID_LINK = '/dev/disk/by-id/usb-PTX_sdmux_HS-SD_MMC_1234-0:0';
+const HW_SERIAL5: Board.SERIAL_PORT_ID = 5;
+const BAUD_RATE = 9600;
+enum GPIO {
+	WRITE_DAC_REG = 0x00,
+	ENABLE_VOUT_SW = 0x03,
+	DISABLE_VOUT_SW = 0x04,
+	ENABLE_VREG = 0x07,
+	ENABLE_FAULTRST = 0x10,
+	SD_RESET_ENABLE = 0x12,
+	SD_RESET_DISABLE = 0x13,
+}
+
+enum PINS {
+	LED_PIN = 13,
+	SD_MUX_SEL_PIN = 28,
+	USB_MUX_SEL_PIN = 29,
+}
+
+/**
+ * Signal handling function
+ */
+async function manageHandlers(
+	handler: (signal: NodeJS.Signals) => Promise<void>,
+	options: { register: boolean },
+): Promise<void> {
+	for (const signal of ['SIGINT', 'SIGTERM'] as Array<NodeJS.Signals>) {
+		if (options.register) {
+			process.on(signal, handler);
+		} else {
+			process.removeListener(signal, handler);
+		}
+	}
+}
+
+interface WorkerOptions {
+	diskDev: string;
+}
+export class TestBot {
+	private board: Board;
+	private options: WorkerOptions;
+	private mutex: Mutex;
+	private signalHandler: (signal: NodeJS.Signals) => Promise<void>;
+
+	/**
+	 * Represents a TestBot
+	 */
+	// Firmata types devicePath as any, will do the same
+	constructor(devicePath: any, options?: WorkerOptions) {
+		this.board = new Board(devicePath);
+
+		this.board.serialConfig({
+			portId: HW_SERIAL5,
+			baud: BAUD_RATE,
+		});
+
+		if (options != null) {
+			this.options = options;
+
+			if (process.platform === 'linux' && this.options.diskDev == null) {
+				throw new Error(
+					'We cannot automatically detect the testbot interface, please provide it manually',
+				);
+			}
+		}
+
+		this.mutex = new Mutex();
+		this.signalHandler = this.disconnect.bind(this);
+	}
+
+	/**
+	 * Critical Section function
+	 */
+	private async criticalSection(
+		section: (args: IArguments) => unknown,
+		args: IArguments,
+	): Promise<void> {
+		const release = await this.mutex.acquire();
+		return Bluebird.try(() => {
+			return Reflect.apply(section, this, args);
+		}).finally(() => {
+			release();
+		});
+	}
+
+	/**
+	 * Get dev interface of the SD card
+	 */
+	private async getDevInterface(
+		timeout: retry.Options = { max_tries: 5, interval: 5000 },
+	): Promise<string> {
+		return retry(() => {
+			return fs.realpath(
+				this.options.diskDev == null ? DEV_ID_LINK : this.options.diskDev,
+			);
+		}, _.assign({ throw_original: true }, timeout));
+	}
+
+	/**
+	 * Send an array of bytes over the selected serial port
+	 */
+	private sendCommand(
+		command: number,
+		settle: number = 0,
+		a: number = 0,
+		b: number = 0,
+	): Bluebird<void> {
+		this.board.serialWrite(HW_SERIAL5, [command, a, b]);
+		return Bluebird.delay(settle);
+	}
```

Comment:
You can make the function `async`, return a `Promise<void>` and change the return to `await Bluebird.delay(settle)`